docs/core/ibc.md

```diff
@@ -0,0 +1,347 @@
+<!--
+order: 9
+-->
+
+# IBC
+
+This document serves as a guide for developers who want to write their own IBC applications. Due to the modular design of the IBC protocol, IBC application developers do not need to concern themselves with the low-level details of clients, connections, and proof verification. Nevertheless a brief explanation of the lower levels of the stack is given so that application developers may have a high-level understanding of the IBC protocol. Then the document goes into detail on the abstraction layer most relevant for application developers (channels and ports), and describes how to define your own custom packets, and `IBCModule` callbacks.
+
+To have your module interact over IBC you must: bind to a port(s), define your own packet data (and optionally acknowledgement) structs as well as how to encode/decode them, and implement the `IBCModule` interface. Below is a more detailed explanation of how to write an IBC application module correctly.
+
+## Pre-requisites Readings
+
+- [Interchain Standards](https://github.com/cosmos/ics) {prereq}
```

Comment:
I don't think we want this to be a pre-requisite reading, it should be possible to use IBC without reading or understanding the full spec.